plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1489,27 +1489,31 @@ protected String deleteSnapshotBackup(final Connection conn, final Long dcId, fi
         return result;
     }
 
-    public void destroyPatchVbd(final Connection conn, final String vmName) throws XmlRpcException, XenAPIException {
+    public void ejectPatchVbd(final Connection conn, final Host host) {
+        try {
+            final Set<VM> vms = host.getResidentVMs(conn);
+            for (final VM vm : vms) {
+                destroyPatchVbd(conn, vm);
+            }
+        } catch (XenAPIException | XmlRpcException ignored) {}
```

Comment:
can we ignore this completely (without at least diagnostic debug message)?

Reply:
Ignorable I think